common/scala/src/main/scala/org/apache/openwhisk/core/database/ActivationStore.scala

```diff
@@ -42,10 +42,18 @@ trait ActivationStore {
    * @param notifier cache change notifier
    * @return Future containing DocInfo related to stored activation
    */
-  def storeAfterCheck(activation: WhiskActivation, context: UserContext)(
-    implicit transid: TransactionId,
-    notifier: Option[CacheChangeNotification]): Future[DocInfo] = {
-    if (context.user.limits.storeActivations.getOrElse(true)) {
+  def storeAfterCheck(activation: WhiskActivation, context: UserContext)(implicit transid: TransactionId,
+                                                                         notifier: Option[CacheChangeNotification],
+                                                                         isBlockingActivation: Boolean = false,
+                                                                         disableStoreResultConfig: Option[Boolean] =
+                                                                           Some(false)): Future[DocInfo] = {
+    if (context.user.limits.storeActivations.getOrElse(true) &&
+        shouldStoreActivation(
+          activation.response.isSuccess,
+          isBlockingActivation,
+          transid.meta.extraLogging,
+          disableStoreResultConfig.get)) {
+
       store(activation, context)
     } else {
       Future.successful(DocInfo(activation.docid))
```

Comment:
Here I would suggest adding a log message with summary of useful activation data, indicating it was successful and explicitly NOT stored. At least: action, namespace, activation id, response size -  but maybe also it would be good if this is a web action to indicate the http status code(I'm not sure how easy that will be to do here, vs elsewhere in the code). This would provide some traceability in the case of "my web action returned a 503 status and I have no idea why" (because your web action chose the 503 status).  WDYT?

Reply:
Yeah good idea. Will add a log with a useful summary. But i suggest making this configurable, as `context.user.limits.storeActivations=false` will trigger this intensive logging too and some people might not want that.

Reply:
Added configurable logging for NOT stored activations: activationId, namespace and response size are part of the log. I think we should differ the http status code for a different PR that logs all activations with an extensive metadata selection.